### ISSUE TITLE
Buildkite agent beta 33 breaks trigger steps

### ISSUE THREAD

**Issue by sander-m** [buildkite/agent#572]
We recently upgraded our agents to 3.0-beta-33, and this has introduced a critical bug.

**Triggered builds no longer build for the specified commit or branch**, but instead build for `master / HEAD` no matter what arguments are passed to the `trigger` step.

This is what our build steps look like:
```yaml
- trigger: "triggered-pipeline"
  build:
    message: ${BUILDKITE_MESSAGE}
    commit: ${BUILDKITE_COMMIT}
    branch: ${BUILDKITE_BRANCH}
```

With beta-33, this always run the build on `master / HEAD` for `triggered-pipeline`, rather than the current commit and branch.

## Why is this important?
Our pipelines are split up into separate pipelines with triggered builds. So this bug rendered our CI pipelines useless. As a result, this bug has caused our developers to merge failing builds to `master`.

**A rollback to beta 31 has fixed it.**

**Comment by lox** [comment]
I'm sorry you ran into this @sander-m. With beta 32 we introduced a pretty major refactor (and extensive tests) for our bootstrap, which has meant the beta's after that have been bumpy. The upshot is that we found a lot of edge cases with the testing and things will get more stable very soon as we prepare for a final release. 

I think this bug was likely caused by an issue in beta 32 where pipeline upload didn't provide environment for interpolation. This is fixed in beta 34. I'm going to put a notice on beta 32 and 33 that they have known critical bugs and should be avoided.
